tests/examples/examples.test.ts

```diff
@@ -0,0 +1,105 @@
+import { spawn } from "child_process";
+import { dirname, basename, resolve } from "path";
+import { Goth } from "../goth/goth";
+import chalk from "chalk";
+import testExamples from "./examples.json";
+
+const noGoth = process.argv[2] === "--no-goth";
+const gothConfig = resolve("../goth/assets/goth-config.yml");
+const gothStartingTimeout = 180;
+const goth = new Goth(gothConfig);
+
+const examples = !noGoth ? testExamples.filter((e) => !e?.noGoth) : testExamples;
+const criticalLogsRegExp = [/Task *. timeot/, /Task *. has been rejected/, /ERROR: TypeError/, /ERROR: Error/gim];
+
+type Example = {
+  cmd: string;
+  path: string;
+  args?: string[];
+  timeout?: number;
+  noGoth?: boolean;
+  skip?: boolean;
+};
+
+async function test(cmd: string, path: string, args: string[] = [], timeout = 120) {
+  const file = basename(path);
+  const cwd = dirname(path);
+  const spawnedExample = spawn(cmd, [file, ...args], { cwd });
+  spawnedExample.stdout?.setEncoding("utf-8");
+  spawnedExample.stderr?.setEncoding("utf-8");
+  let error = "";
+  const timeoutId = setTimeout(() => {
+    error = `Test timeout was reached after ${timeout} seconds.`;
+    spawnedExample.kill();
+  }, timeout * 1000);
+  return new Promise((res, rej) => {
+    spawnedExample.stdout?.on("data", (data: string) => {
+      console.log(data.trim());
+      const logWithoutColours = data.replace(
+        /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g,
+        "",
+      );
+      if (criticalLogsRegExp.some((regexp) => logWithoutColours.match(regexp))) {
+        error = `A critical error occurred during the test.`;
```

Comment:
why not use the actual error message here instead of the generic `critical error occurred`?

Reply:
you're right, i could add that